### DISCUSSION TITLE
How to delete bulk content and overcome the limit

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Knowlg/Community#182] [General Discussions]
Hello Everyone,

We are  upgrading sunbird ed instance , release 5.0 to release 5.1 we want to minimise the time for migration in order to do so we are planning to Retire & delete content in bulk about 2 Lakh + content.
We would like to delete the content so that migration is done faster.
Can you suggest how do we do this ?

We are thinking to retire content but
-  When we run the api to get all the contents the api limits to 10,000 hence we are not able to get more data.
- When we retire does it remove content from neo4j, kasandra , elastic search ?
- If not can you suggest how do we clean up these record

**Comment by pallakartheekreddy** [comment]
@siddhishinde0723 Currently, We don't have any bulk delete/retire tools or API.
In our systems, when we retire an object (such as content or a collection) using the retire API, it performs a soft delete. This means it changes the status of the object to `Retired` but does not remove it from Neo4j, Cassandra, and Elasticsearch.

Yes, the Search API will return only 10,000 records in response. As you already know, there are 2 Lakh + contents, please consider using specific filters to get those content IDs.

**Reply by siddhishinde0723** [reply]
@pallakartheekreddy Thank you for the update
